sw/device/boot_rom/bootstrap.c

```diff
@@ -64,17 +64,54 @@ static int erase_flash(void) {
   return 0;
 }
 
+/**
+ * Computes the SHA256 of the given data.
+ */
+static void compute_sha256(const dif_hmac_t *hmac, const void *data, size_t len,
+                           dif_hmac_digest_t *digest) {
+  CHECK(dif_hmac_mode_sha256_start(hmac) == kDifHmacOk);
+
+  const uint8_t *data8 = data;
+  size_t data_left = len;
+  while (data_left > 0) {
+    size_t bytes_sent;
+    switch (dif_hmac_fifo_push(hmac, data8, data_left, &bytes_sent)) {
+      case kDifHmacFifoOk:
+      case kDifHmacFifoFull:
+        data8 += bytes_sent;
+        data_left -= bytes_sent;
```

Comment:
This falls through. I don't like the loop-switch control flow here, but like the loop-switch-goto less from the other commit, can you make this more vanilla flow control?

Reply:
Regardless of how I rewrite it, I'm surprised we don't have `-Wimplicit-fallthrough` turned on. We should absolutely turn it on.